### PR TITLE
docs: Improve BUILD.md

### PR DESCRIPTION
**Pull request by juan-lunarg**
- Address concerns from PR#4659
- Use bash syntax highlighting
- Update Visual Studio docs

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -349,8 +349,8 @@ Also CMAKE_EXPORT_COMPILE_COMMANDS is implemented only by Makefile and Ninja gen
   - Any Personal Computer version supported by Microsoft
 - Microsoft [Visual Studio](https://www.visualstudio.com/)
   - Versions
-    - [2015](https://www.visualstudio.com/vs/older-downloads/)
-    - [2017](https://www.visualstudio.com/vs/downloads/)
+    - [2022](https://www.visualstudio.com/vs/downloads/)
+    - [2015-2019](https://www.visualstudio.com/vs/older-downloads/)    
   - The Community Edition of each of the above versions is sufficient, as
     well as any more capable edition.
 - [CMake 3.10.2](https://cmake.org/files/v3.10/cmake-3.10.2-win64-x64.zip) is recommended.
```
Comment (lunarpapillo) [inline BUILD.md:356]: Unrelated to your changes, but this caught my eye now.  I believe we currently require a minimu of **CMake** 3.10.2, but I *think* our documented "recommended" **CMake** version is supposed to match the version we actually use in CI, right?  (That would currently be 3.19.3.)

If that's true, this text should be updated to include both the minimum and recommended **CMake** versions, e.g.:
```
- [CMake 3.10.2](https://cmake.org/files/v3.10/cmake-3.10.2-win64-x64.zip) is the minimum CMake version supported.  [CMake 3.19.3](https://cmake.org/files/v3.19/cmake-3.19.3-win64-x64.zip) is recommended.
```

File: BUILD.md [hunk 2]
```diff
@@ -411,13 +411,16 @@ create a build directory and generate the Visual Studio project files:
 The `-A` option is used to select either the "Win32" or "x64" architecture.
 
 If a generator for a specific version of Visual Studio is required, you can
-specify it for Visual Studio 2015, for example, with:
+specify it for Visual Studio 2022, for example, with:
```
Comment (lunarpapillo) [inline BUILD.md:414]: The sentence here is awkward.  Replace with:
```
specify it with the `-G` switch.  For example, to specify Visual Studio 2022, use:
```

File: BUILD.md [hunk 3]
```diff
@@ -411,13 +411,16 @@ create a build directory and generate the Visual Studio project files:
...
 
-    64-bit: -G "Visual Studio 14 2015 Win64"
-    32-bit: -G "Visual Studio 14 2015"
+    64-bit: -G "Visual Studio 14 2022"
+    32-bit: -G "Visual Studio 14 2022" -A Win32
```
Comment (lunarpapillo) [inline BUILD.md:417]: I believe VS2022 would be: `-G "Visual Studio 17 2022"` in both examples.  ("14" is the revision number for VS2015.)

File: BUILD.md [hunk 4]
```diff
@@ -411,13 +411,16 @@ create a build directory and generate the Visual Studio project files:
...
 
-See this [list](#cmake-visual-studio-generators) of other possible generators
-for Visual Studio.
+NOTE:
+By default VS2019 and higher target "x64"
+By default VS2017 and lower target "Win32"
```
Comment (lunarpapillo) [inline BUILD.md:421]: Turn this into a list (or add punctuation, so they won't be sentence fragments), e.g.:
```
NOTE:
- By default, VS2019 and higher target `x64`
- By default, VS2017 and lower target `Win32`

File: BUILD.md [hunk 5]
```diff
@@ -703,18 +695,22 @@ you are building for.
...
-    export ASFLAGS=--32
-    export CFLAGS=-m32
-    export CXXFLAGS=-m32
-    export PKG_CONFIG_LIBDIR=/usr/lib/i386-linux-gnu
+```bash
+export ASFLAGS=--32
+export CFLAGS=-m32
+export CXXFLAGS=-m32
+export PKG_CONFIG_LIBDIR=/usr/lib/i386-linux-gnu
+```
```
Comment (lunarpapillo) [inline BUILD.md:703]: Do we also need:
```
export PKG_CONFIG_PATH=/usr/lib/i386-linux-gnu-pkgconfig
```
?

File: BUILD.md [hunk 6]
```diff
@@ -757,11 +753,13 @@ version, as Android Studio will roll it forward fairly regularly.
...
-    export ANDROID_NDK_HOME=$HOME/Android/sdk/ndk-bundle
-    export PATH=$ANDROID_SDK_HOME:$PATH
-    export PATH=$ANDROID_NDK_HOME:$PATH
-    export PATH=$ANDROID_SDK_HOME/build-tools/26.0.3:$PATH
+```bash
+export ANDROID_SDK_HOME=$HOME/Android/sdk
+export ANDROID_NDK_HOME=$HOME/Android/sdk/ndk-bundle
+export PATH=$ANDROID_SDK_HOME:$PATH
+export PATH=$ANDROID_NDK_HOME:$PATH
+export PATH=$ANDROID_SDK_HOME/build-tools/26.0.3:$PATH
```
Comment (lunarpapillo) [inline BUILD.md:761]: The `build-tools/N.N.N` directory will change for more recent Android installations (for reference, `tcubuand` and `leasath` have `build-tools/29.0.2`).  CI happens to search the `build-tools` directory looking for the most recent version.

I'm sure you can find a clever way to say this. :-)

File: BUILD.md [hunk 7]
```diff
@@ -771,10 +769,12 @@ On Windows:
...
 
-    export ANDROID_SDK_HOME=$HOME/Library/Android/sdk
-    export ANDROID_NDK_HOME=$HOME/Library/Android/sdk/ndk-bundle
-    export PATH=$ANDROID_NDK_HOME:$PATH
-    export PATH=$ANDROID_SDK_HOME/build-tools/26.0.3:$PATH
+```bash
+export ANDROID_SDK_HOME=$HOME/Library/Android/sdk
+export ANDROID_NDK_HOME=$HOME/Library/Android/sdk/ndk-bundle
+export PATH=$ANDROID_NDK_HOME:$PATH
+export PATH=$ANDROID_SDK_HOME/build-tools/26.0.3:$PATH
```
Comment (lunarpapillo) [inline BUILD.md:776]: Probably similar to above.

**Review by lunarpapillo — CHANGES_REQUESTED** [review]
Mostly style.  Should fix the VS2022 version number, though.

Have you tried the command lines to make sure they work? ;-)

File: BUILD.md (hunk 5 above)
Comment (juan-lunarg) [reply BUILD.md:703]: Just tested the 32 bit build. Doesn't appear `PKG_CONFIG_PATH` is needed.

File: BUILD.md (hunk 3 above)
Comment (juan-lunarg) [reply BUILD.md]: Good catch you are correct.

**Comment by juan-lunarg** [comment]
> Have you tried the command lines to make sure they work? ;-)

I did test the 32 bit linux. I got trigger happy with WIN32 due to having documented similar things in the past, and my overconfidence gets to me.

File: BUILD.md [hunk 8]
```diff
@@ -411,13 +411,16 @@ create a build directory and generate the Visual Studio project files:
...
+    64-bit: -G "Visual Studio 17 2022"
+    32-bit: -G "Visual Studio 17 2022" -A Win32
 
-See this [list](#cmake-visual-studio-generators) of other possible generators
-for Visual Studio.
+NOTE:
+- By default VS2019 and higher target "x64"
+- By default VS2017 and lower target "Win32"
+
+Check the official CMake docs for further information on Visual Studio generators
```
Comment (lunarpapillo) [inline BUILD.md:423]: I think I'd made an earlier comment on this line that was missed...

File: BUILD.md [hunk 9]
```diff
@@ -757,11 +753,13 @@ version, as Android Studio will roll it forward fairly regularly.
...
-    export ANDROID_NDK_HOME=$HOME/Android/sdk/ndk-bundle
-    export PATH=$ANDROID_SDK_HOME:$PATH
-    export PATH=$ANDROID_NDK_HOME:$PATH
-    export PATH=$ANDROID_SDK_HOME/build-tools/26.0.3:$PATH
+```bash
+export ANDROID_SDK_HOME=$HOME/Android/sdk
+export ANDROID_NDK_HOME=$HOME/Android/sdk/ndk-bundle
+export PATH=$ANDROID_SDK_HOME:$PATH
+export PATH=$ANDROID_NDK_HOME:$PATH
+export PATH=$ANDROID_SDK_HOME/build-tools/X.Y.Z:$PATH
```
Comment (lunarpapillo) [inline BUILD.md:761]: Add a comment explaining what `X.Y.Z` needs to be, e.g.:

> where `X.Y.Z` is the most recent version number inside the `build-tools` directory.

**Review by lunarpapillo — CHANGES_REQUESTED** [review]

File: BUILD.md (hunk 8 above)
Comment (lunarpapillo) [reply BUILD.md:423]: Hmmm, but I can't find any reference to it... I recommend turning this into a sentence (without a period, it's just a fragment) and including a link, e.g.:
```
Check the [official CMake documentation](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#visual-studio-generators) for further information on Visual Studio generators.
```

**Review by lunarpapillo — APPROVED** [review]
Thanks for your patience and persistence. :-)  This is much better than it was.

**Pull request merged**